bouncing-balls/index.js

```diff
@@ -47,6 +47,42 @@ while (ballsPool.length < ballsCount) {
   ballsPool.push(ball);
 }
 
+const resizableBall = {
+  x: generateRandomNumber(0 + ballSize, canvasWidth - ballSize),
+  y: generateRandomNumber(0 + ballSize, canvasHeight - ballSize),
+  velocityX: generateRandomNumber(-maxVelocity, maxVelocity),
+  velocityY: generateRandomNumber(-maxVelocity, maxVelocity),
+  color: 'rgb(' + generateRandomNumber(0, 255) + ',' + generateRandomNumber(0, 255) + ',' + generateRandomNumber(0, 255) + ')',
+  size: ballSize,
+  draw() {
+    canvasContext.beginPath();
+    canvasContext.fillStyle = this.color;
+    canvasContext.arc(this.x, this.y, this.size, 0, 2 * Math.PI);
+    canvasContext.fill();
+  },
+  update() {
+    const intersectVerticalBorder = (this.x + this.size) >= canvasWidth
+      || (this.x - this.size) <= 0;
+    const intersectHorizontalBorder = (this.y + this.size) >= canvasHeight
+      || (this.y - this.size) <= 0;
+
+    if (intersectVerticalBorder) {
+      this.velocityX = -(this.velocityX);
+    }
+
+    if (intersectHorizontalBorder) {
+      this.velocityY = -(this.velocityY);
+    }
+
+    this.x = this.x + this.velocityX;
+    this.y = this.y + this.velocityY;
```

Comment:
Seems like a lot of code duplication, let's try to organize the code in a better way